docs/ocean/design_docs/vert-lagrang-remap/vert-lagrang-remap.rst

```diff
@@ -0,0 +1,524 @@
+safeguards to prevent layers from becoming too thin or deformed. 
+
+Requirement: Ability to specify remapping method and its options
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/04
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+There are several choices to make for the remapping operation to balance accuracy, 
+stability and computational cost. To the extent that it is feasible, the user 
+should have the ability to control the order of accuracy of the remapping method,
+[Darren, other options here?] and other remapping options (e.g., via namelist options).
+
+Requirement: Support for existing time stepping schemes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/04
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+Vertical Lagrangian remapping should be supported (eventually) in both the RK4 
+and the split-explicit time stepping schemes currently implemented in MPAS-Ocean. 
+
+Requirement: not climate changing
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/15
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+For the same target grid, time-stepping scheme and problem, the solution for 
+layer thicknesses and key prognostic variables using VLR is within a non-
+climate-changing threshold value [presumably greater than machine precision?] 
+of the solution when VLR is not performed.
+
+Requirement: performance
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/04
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+Performance is not degraded below an acceptable level *which is to be 
+determined*.
+
+Requirement: no interference
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/04
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+No calculations are made outside of remapping itself using prognostic (or 
+diagnostic) variables both before and after remapping to avoid introducing 
+additional errors.
+
+Requirement: modularity
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/15
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+To the degree possible, the code for determining the target grid and performing 
+remapping should be kept in its own Fortran module(s) for better readability.
+
+
+
+Algorithm Design (optional)
+---------------------------
+
+Algorithm Design: Ability to perform vertical Lagrangian-remapping
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/15
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+The conservation of momentum, volume, and tracer equations in MPAS-Ocean are:
+
+.. math::
+
+   \frac{\partial u_k}{\partial t} + q_k h_k u_k^{normal} + \overline{w^t \delta z^t u} = -\frac{1}{\rho_0} \nabla p_k - \frac{rho_k g}{\rho_0} \nabla z_k - \nabla K_k + [D_h^u]_k + [D_{\nu}^u]_k + F_k^u
```

Comment:
```suggestion
   \frac{\partial u_k}{\partial t} + q_k h_k u_k^{normal} + \overline{w^t \delta z^t u} = -\frac{1}{\rho_0} \nabla p_k - \frac{\rho_k g}{\rho_0} \nabla z_k - \nabla K_k + [D_h^u]_k + [D_{\nu}^u]_k + F_k^u
```

Reply:
I personally find the square bracket notation that @mark-petersen used on the `D_h` and `D_v` terms to be confusing because those square brackets are used in Ringler et al. (2013) to mean a vertical average, whereas here they are just used to handle the fact that there is already a subscript (`h` or `v`) on the terms.

Reply:
My suggestion would be `D_{h,k}^u`, etc.

Reply:
I take that back. After a closer look, I think Ringler et al. (2013) is using square brackets in the same way as Petersen et al. (2014), so it's fine as it is.